src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.AzureServiceBus
+{
+    public class AzureServiceBusSubscriptionHealthCheck : IHealthCheck
+    {
+        private static readonly ConcurrentDictionary<string, ManagementClient> ManagementClientConnections =
+            new ConcurrentDictionary<string, ManagementClient>();
+        private readonly string _connectionString;
+        private readonly string _topicName;
+        private readonly string _subscriptionName;
+
+
+        public AzureServiceBusSubscriptionHealthCheck(string connectionString, string topicName, string subscriptionName)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new ArgumentNullException(nameof(connectionString));
+            }
+
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException(nameof(topicName));
+            }
+
+            if (string.IsNullOrEmpty(subscriptionName))
+            {
+                throw new ArgumentNullException(nameof(subscriptionName));
+            }
+
+            _connectionString = connectionString;
+            _topicName = topicName;
+            _subscriptionName = subscriptionName;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var connectionKey = $"{_connectionString}_{_topicName}_{_subscriptionName}";
+                if (!ManagementClientConnections.TryGetValue(connectionKey, out var managementClient))
+                {
+
+                    managementClient = new ManagementClient(_connectionString);
```

Comment:
There is a possibility of double creation `ManagementClient`.

Reply:
hmm, can you elaborate why it would be possible?

Reply:
Because more than one thread can hit that line at the same time. I wouldn't consider such a scenario if you didn't use `ConcurrentDictionary`. What is the other reason for using it?

Reply:
I would not worry of having ManagementClient created twice. It's a local variable and would be garbage-collected soon if it happens. What matters is the instance is added only once to the ManagementClientConnections, I suppose.

Reply:
It just **might** be. If that's not a problem, then it doesn't matter.